JBTM-2968 Some JTS quickstarts run with JacORB

marcosgopen opened the pull request:
https://issues.redhat.com/browse/JBTM-2968

Setting Open JDKOrb as default ORB.
When working I left the JacOrb implementation as an alternative.
Updated some READMEs for clarity

jmfinelli commented on ArjunaJTS/recovery/run.sh:
```diff
@@ -22,16 +22,16 @@ set -m
 
 # if you want to test using the sun orb you 
-# SUN_CONIG="-Dcom.sun.CORBA.POA.ORBServerId=1 -Dcom.sun.CORBA.POA.ORBPersistentServerPort=12567"
+SUN_CONIG="-Dcom.sun.CORBA.POA.ORBServerId=1 -Dcom.sun.CORBA.POA.ORBPersistentServerPort=12567"
```
Same here

jmfinelli commented on ArjunaJTS/standalone/pom.xml:
```diff
@@ -168,6 +171,63 @@
+                      <key>QUICKSTART_NARAYANA_VERSION</key>
+                      <value>${version.narayana}</value>
+                    </systemProperty>
+                    <systemProperty>
+                      <key>OrbPortabilityEnvironmentBean.orbImpleClassName</key>
+                      <value>com.arjuna.orbportability.internal.orbspecific.javaidl.orb.implementations.javaidl_1_4</value>
+                    </systemProperty>
+                    <systemProperty>
+                      <key>OrbPortabilityEnvironmentBean.poaImpleClassName</key>
+                      <value>com.arjuna.orbportability.internal.orbspecific.javaidl.oa.implementations.javaidl_1_4</value>
+                    </systemProperty>
+                    <systemProperty>
+                      <key>OrbPortabilityEnvironmentBean.orbDataClassName</key>
+                      <value>com.arjuna.orbportability.internal.orbspecific.versions.javaidl_1_4</value>
+                    </systemProperty>
```
See first comment

jmfinelli commented on ArjunaJTS/standalone/start.sh:
```diff
@@ -53,9 +53,9 @@ function startRM {
 function startClient {
   if [ $ORBTYPE = "jacorb" ]; then
-    mvn compile exec:java  -Dexec.cleanupDaemonThreads=false -Dexec.mainClass=org.jboss.narayana.jta.quickstarts.RemoteTMExample -DOrbPortabilityEnvironmentBean.orbImpleClassName=com.arjuna.orbportability.internal.orbspecific.jacorb.orb.implementations.jacorb_2_0 -DNAME_SERVER_HOST="$HOST_ADDRESS"
+    mvn compile exec:java  -Pscript -Dexec.cleanupDaemonThreads=false -Dexec.mainClass=org.jboss.narayana.jta.quickstarts.RemoteTMExample -DOrbPortabilityEnvironmentBean.orbImpleClassName=com.arjuna.orbportability.internal.orbspecific.jacorb.orb.implementations.jacorb_2_0 -DNAME_SERVER_HOST="$HOST_ADDRESS"
```
Same here. This `if` should not be needed as we have only one possible choice (i.e. openjdk-orb)

tomjenkinson replied:
An aside that I think it may be possible we would work with `ibmorb` for IBM JDK 11 still (assuming IBM's JDK 11)  has one but I don't know - I don't think it's present in IBM JDK 17. That being said it doesn't look like https://github.com/jbosstm/quickstart/blob/master/ArjunaJTS/standalone/start.sh has reference to ibmorb anyway.

tomjenkinson commented on ArjunaJTS/standalone/README.md:
```diff
@@ -68,7 +68,7 @@ Running with JacOrb:
 Running with JdkORB:
 
-as above but replace JacORB with JdkORB
+as above but replace jacorb with jdkorb
```
The JDK ORB is no longer supported by Narayana and has been removed with work on JBTM-3590 as the minimum Java version is now 11 and given JEP-320 then we no longer should try to work with the JDK ORB.

That's a bit of pre-amble, but the overall point is rather than JDKORB we should be referring to OpenJDK ORB (it's a JBoss fork of that actually: https://github.com/jboss/openjdk-orb)

tomjenkinson commented on ArjunaJTS/trailmap/pom.xml:
```diff
@@ -1,122 +1,278 @@
+        </executions>
+      </plugin>
+    </plugins>
+    <pluginManagement>
+      <plugins>
+        <!--This plugin's configuration is used to store Eclipse m2e settings 
+          only. It has no influence on the Maven build itself. -->
+        <plugin>
+          <groupId>org.eclipse.m2e</groupId>
+          <artifactId>lifecycle-mapping</artifactId>
+          <version>1.0.0</version>
+          <configuration>
+            <lifecycleMappingMetadata>
+              <pluginExecutions>
+                <pluginExecution>
+                  <pluginExecutionFilter>
+                    <groupId>org.codehaus.mojo</groupId>
+                    <artifactId>idlj-maven-plugin</artifactId>
+                    <versionRange>[1.2.1,)</versionRange>
+                    <goals>
+                      <goal>generate</goal>
+                    </goals>
+                  </pluginExecutionFilter>
+                  <action>
+                    <ignore></ignore>
+                  </action>
+                </pluginExecution>
+              </pluginExecutions>
+            </lifecycleMappingMetadata>
+          </configuration>
+        </plugin>
+      </plugins>
+    </pluginManagement>
```
I don't think it's strictly necessary: https://github.com/jbosstm/quickstart/commit/4f9588a59cc583a80372e8a0605fb9ce5d33f530

I think it looks like it is intended to ignore the IDL compiler if you import it the project to Eclipse - whether it still works I don't know.

tomjenkinson commented on ArjunaJTS/recovery/run.bat:
```diff
@@ -18,16 +19,16 @@ rem MA  02110-1301, USA.
 echo "Running JTS recovery quickstart"
 
-rem SUN_CONIG="-Dcom.sun.CORBA.POA.ORBServerId=1 -Dcom.sun.CORBA.POA.ORBPersistentServerPort=12567"
+SUN_CONIG="-Dcom.sun.CORBA.POA.ORBServerId=1 -Dcom.sun.CORBA.POA.ORBPersistentServerPort=12567"
```
+1

tomjenkinson commented on ArjunaJTS/pom.xml:
```diff
@@ -129,7 +142,7 @@
       <activation>
         <property>
-          <name>!useJdkOrb</name> 
+          <name>usejacOrb</name> 
```
Removing the jacorb profile seems right: https://github.com/jbosstm/quickstart/blob/master/ArjunaJTS/pom.xml#L128-L142

tomjenkinson commented:
TESTIT

tomjenkinson commented on ArjunaJTS/pom.xml:
```diff
@@ -53,7 +53,22 @@
+          <version>3.0.0</version>
+          <configuration>
+           
+          <systemProperty>
```
It would be good if these were no longer required. Is the main codebase in that state yet?

tomjenkinson commented on ArjunaJTS/jts/pom.xml:
```diff
@@ -82,8 +82,8 @@
                 <configuration>
-                    <source>1.6</source>
-                    <target>1.6</target>
+                    <source>11</source>
```
@marcosgopen please can you put this in a JBTM-3590 commit?

tomjenkinson commented on ArjunaJTS/pom.xml:
```diff
@@ -89,8 +87,8 @@
         <configuration>
-          <source>1.6</source>
-          <target>1.6</target>
+          <source>11</source>
```
Please associate the JDK11 work with a JBTM-3590 commit

tomjenkinson commented:
@marcosgopen please can you consider taking some of the commits from: https://github.com/jbosstm/quickstart/pull/398/commits into your pull request?

In particular the ones that are prefixed JBTM-3590 (there are two currently), JBTM-3597, JBTM-3482. In particular you shouldn't need "CI SCRIPT: GitHub Action" or "Use bash to get function" (although this latter one should definitely go in at some point so you are welcome to take it too - the shell is defined as /bin/sh but uses bash notation to define the function).

tomjenkinson commented:
@marcosgopen FYI I opened https://github.com/marcosgopen/quickstart/pull/3

marcosgopen commented:
Yes @tomjenkinson , I will reply in more detail [here](https://github.com/marcosgopen/quickstart/pull/3)

tomjenkinson commented:
@marcosgopen I think the failures in CI that are related to LRA are related to the (partial) upgrade to Quarkus. I think actually they might be related to something that @xstefank was talking recently about in regards to user configured LRA deployments (https://narayana.zulipchat.com/#narrow/stream/323714-users/topic/LRA.20not.20working.20in.20WildFly.20when.20deployed.20manually). I still think we should upgrade to the Quarkus version as soon as possible. We could consider to merge this pull and complete the update to Quarkus for the quickstarts soon after (pre-release) or we could decide to finish the upgrade here and therefore not introduce those failing parts post merge /cc @mmusgrov

marcosgopen commented:
> @marcosgopen I think the failures in CI that are related to LRA are related to the (partial) upgrade to Quarkus. I think actually they might be related to something that @xstefank was talking recently about in regards to user configured LRA deployments (https://narayana.zulipchat.com/#narrow/stream/323714-users/topic/LRA.20not.20working.20in.20WildFly.20when.20deployed.20manually). I still think we should upgrade to the Quarkus version as soon as possible. We could consider to merge this pull and complete the update to Quarkus for the quickstarts soon after (pre-release) or we could decide to finish the upgrade here and therefore not introduce those failing parts post merge /cc @mmusgrov

Thanks @tomjenkinson for your comment. Even if I don't like to leave some unfixed quickstarts behind I agree with you. I would suggest to disable temporary the 3 quickstarts that are not working, create a jira for each of them and merge this PR.
This are the 3 quickstarts that need a fix:
- examples-microprofile-lra-cdi-embedded (https://issues.redhat.com/browse/JBTM-3649);
-  jta-and-hibernate-standalone ( a transaction commit should fail and it does not);
-  examples-microprofile-lra-cdi-participant (I still have to figure it out, it works locally:  java.lang.NoClassDefFoundError: io/narayana/lra/AnnotationResolver);

tomjenkinson commented:
Regarding the LRA and Quarkus problem(s?), may be somehow related to https://github.com/quarkusio/quarkus/issues/25330 / https://github.com/quarkusio/quarkus/pull/26155. Perhaps we need to add some dependency on the extension to the quickstart? /cc @xstefank - what do you think?

xstefank commented:
@tomjenkinson I am not diving into the whole PR, but if the error is only what @marcosgopen wrote in his last comment for the examples-microprofile-lra-cdi-participant, then no, this is not related. These PRs you linked are fixes for narayana-lra with resteasy-reactive backend which I don't think we use in the quickstart. But if I'm wrong I'll take a deeper look later, just let me know.

marcosgopen commented:
@tomjenkinson the examples-microprofile-lra-cdi-participant quickstart error gets fixed adding those dependencies: lra-service-base and resteasy-client (even if we still have quarkus-resteasy, that does not include the needed org.glassfish.jersey.client.JerseyClientBuilder). See (temporary) commit https://github.com/jbosstm/quickstart/pull/395/commits/2d5070b84768eff683664aa520e935f5c02a7f9a
Do you think this is  related to https://github.com/quarkusio/quarkus/issues/25330 / https://github.com/quarkusio/quarkus/pull/26155?

tomjenkinson commented:
@xstefank @marcosgopen I thought I had seen something about it looking for Jersey ClientBuilder someplace. I will take a look in the logs see if they show something

tomjenkinson commented:
@marcosgopen sorry, I was too quick before. I think I connected it before through JerseyClientBuilder to those Quarkus topics but it seems you fixed that with the added dependencies. Why do you refer to https://github.com/jbosstm/quickstart/commit/9195d0479a422c31cc81ceefeec13497d0e1d68d as temporary? Is it just the formatting of the commit that you will squash in? I suggest to consider if it should be squashed it into the Quarkus upgrade one (if that is what is triggering the need to add it). But also please can you put the version (https://github.com/jbosstm/quickstart/blob/9195d0479a422c31cc81ceefeec13497d0e1d68d/rts/lra-examples/cdi-participant/pom.xml#L107) into the properties section: https://github.com/jbosstm/quickstart/blob/9195d0479a422c31cc81ceefeec13497d0e1d68d/rts/lra-examples/cdi-participant/pom.xml#L13 ?

tomjenkinson commented:
I don't know if you noticed it but there are 404 HTTP problems:

```+ '[' '' = true ']'
+ echo '===== testing service'
===== testing service
+ test_service '1 completed and 0 compensated'
+ curl -4 -X PUT -I http://localhost:8082/cdi
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
HTTP/1.1 404 Not Found
Narayana-LRA-API-version: 1.0
content-length: 0

++ timeout_adjust 1
+ sleep 1
++ curl -4 http://localhost:8082/cdi
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
+ '[' '' = '1 completed and 0 compensated' ']'
+ return 1
+ res=1
+ test_recovery
+ echo '===== CdiBasedResource halt the service on pid 3813'
===== CdiBasedResource halt the service on pid 3813
+ curl -4 -X PUT -I 'http://localhost:8082/cdi?fault=haltcdiduring'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
HTTP/1.1 404 Not Found
Narayana-LRA-API-version: 1.0
content-length: 0```

tomjenkinson commented:
@marcosgopen I guess you will need to do something like add artifacts to https://github.com/jbosstm/quickstart/blob/master/rts/lra-examples/cdi-embedded/pom.xml too?

xstefank commented:
@tomjenkinson If ClientBuilder looks for Jersey implementation that is JAX-RS client error which requires service file pointing to the correct RESTEasy impl of the ClientBuilder. This file is included in this dependency, I suppose - https://github.com/jbosstm/quickstart/commit/9195d0479a422c31cc81ceefeec13497d0e1d68d#diff-b74811c056ae57129e01c49cdec20a17c8b7ab25b12fc09d826b893a0c8a2cebR104-R108 which is what is fixing your problem. 

@marcosgopen, so I took a look and you are using Quarkus in the narayana quickstart cdi-participant - https://github.com/jbosstm/quickstart/blob/master/rts/lra-examples/cdi-participant/pom.xml. In Quarkus, there is already a narayana-lra extension that bundles narayana-lra, but I can understand why you want to include it manually for quickstart testing here. 

But, if you would try to use the quarkus-narayana-lra extension in quarkus - https://quarkus.io/guides/lra#configuration you can see there that narayana-lra needs to be complemented by both JAX-RS server and client implementation, either classic blocking RESTEasy (quarkus-resteasy-jackson, quarkus-rest-client-jackson) or RESTEasy reactive (quarkus-resteasy-reactive-jackson, quarkus-rest-client-reactive-jackson). -jackson is there because we also need to marshal responses both server and client as JSONs but these extensions bundle underlying extensions without -jackson. So you see, your problem here is that you only included quarkus-resteasy, which is only the server (without JSON), but more importantly, you didn't include the JAX-RS client extension, which is quarkus-rest-client(-reactive)(-jackson) which would include the service file I talk about above.

marcosgopen commented:
@tomjenkinson :
- Thanks for you suggestions and review
- temporary because as we discussed before these quarkus issues should be part of another PR: see https://github.com/jbosstm/quickstart/pull/399 . I answer your questions there.
- I removed my last commit here and moved in the PR https://github.com/jbosstm/quickstart/pull/399
@xstefank :
- Thanks for your review and helpful explanation about  https://quarkus.io/guides/lra#configuration
- I reply to your comment here https://github.com/jbosstm/quickstart/pull/399

marcosgopen commented:
We had an error in jta-and-hibernate-standalone. Byteman did not trigger actions specified here https://github.com/jbosstm/quickstart/blob/master/jta-and-hibernate-standalone/src/test/java/org/jboss/narayana/quickstart/jta/TestCase.java#L157
I upgraded the byteman version to 4.0.19. Now it should work as expected.

tomjenkinson commented:
Thank you @marcosgopen for the fix for the `jta-and-hibernate-standalone` quickstart! Are you able to suggest why you think the byteman needed upgrading? The failure was not caused by Quickstarts changing as a JDK 11 build with fcc51e431f62a22b2e9e8474d1a08becb4677502 passed on 24 Jun 2022, 23:00:10 (UTC+1 I presume) and the same sha failed on 8 Jul 2022, 13:48:49 (again, UTC+1 I presume).

marcosgopen commented:
Sorry @tomjenkinson I am not sure I understood your point: can you link me the successful build job?
> why you think the byteman needed upgrading
I am not sure @tomjenkinson , probably the 3.0.3 version is in conflict with some new narayana transitive dependencies. I can see that upgrading byteman the behaviour of the test is the same as it was the second of April:
- http://narayanaci1.eng.hst.ams2.redhat.com/job/btny-pulls-narayana-quickstart-matrix/jdk=jdk11.latest,label=linux/2/artifact/jta-and-hibernate-standalone/target/surefire-reports/org.jboss.narayana.quickstart.jta.TestCase-output.txt
-  http://narayanaci1.eng.hst.ams2.redhat.com/job/btny-pulls-narayana-quickstart-matrix/jdk=jdk11.latest,label=linux/32/artifact/jta-and-hibernate-standalone/target/surefire-reports/org.jboss.narayana.quickstart.jta.TestCase-output.txt

tomjenkinson commented:
I have sent you the links to the passing and failing jobs for fcc51e431f62a22b2e9e8474d1a08becb4677502. Perhaps it was the changes in Narayana that took place over the time period which caused the rule to stop firing but it's hard to guess why - I looked at the rule and the helper and neither look particularly strange - do we have any byteman logging that could explain why (I guess) the rule is not installed without the upgrade. Anyway, it's great to know your change gets the test passing again.

marcosgopen commented:
> I have sent you the links to the passing and failing jobs for [fcc51e4](https://github.com/jbosstm/quickstart/commit/fcc51e431f62a22b2e9e8474d1a08becb4677502). Perhaps it was the changes in Narayana that took place over the time period which caused the rule to stop firing but it's hard to guess why - I looked at the rule and the helper and neither look particularly strange - do we have any byteman logging that could explain why (I guess) the rule is not installed without the upgrade. Anyway, it's great to know your change gets the test passing again.

I think this is the proper occasion to ask the following:
- Do we have a script that having a parameter (i.g. a commit sha of narayana repo) install that specific narayana project and run the quickstarts against that?

tomjenkinson commented on ArjunaJTS/standalone/pom.xml:
```diff
@@ -129,6 +118,7 @@
                       <key>QUICKSTART_NARAYANA_VERSION</key>
                       <value>${version.narayana}</value>
                     </systemProperty>
+                    
```
formatting, please can you remove?

tomjenkinson commented on ArjunaJTS/standalone/pom.xml:
```diff
@@ -187,23 +178,25 @@
+                <phase>integration-test</phase>
+                <configuration>
+                  <target>
+                    <exec dir="${project.basedir}"
```
I think the changes prior to this line and from line 193 are just formatting and could be reverted

tomjenkinson replied:
I think actually the whole of this section might be a reformat?

tomjenkinson commented on jta-and-hibernate-standalone/src/test/java/org/jboss/narayana/quickstart/jta/TestCase.java:
```diff
@@ -132,7 +137,7 @@ public void testCommit() throws Exception {
     /**
      * Adds two entries to the database and rolls back the transaction. At the end of the test no entries should be in the
      * database.
-     * 
```
formatting, please revert

tomjenkinson commented on jta-and-hibernate-standalone/src/test/java/org/jboss/narayana/quickstart/jta/TestCase.java:
```diff
@@ -166,6 +171,9 @@ public void testCrashBeforeCommit() throws Exception {
             transactionManager.commit();
             fail("QuickstartRuntimeException expected");
         } catch (Throwable t) {
+            if(!(t.getCause() instanceof QuickstartRuntimeException)) {
```
I think we should change to doing this with a message in the assertTrue such as:

`assertTrue("Cause was: " + t.getClass() + ":" + t.getMessage(), t.getCause() instanceof QuickstartRuntimeException);`

tomjenkinson commented:
> * Do we have a script that having a parameter (i.g. a commit sha of narayana repo) install that specific narayana project and run the quickstarts against that?

I think you might be able to set NY_BRANCH and QUICKSTART_NARAYANA_VERSION environment variables to get that kind of thing: https://github.com/jbosstm/quickstart/blob/master/scripts/hudson/quickstart.sh#L100 and https://github.com/jbosstm/quickstart/blob/master/scripts/hudson/quickstart.sh#L163

tomjenkinson commented on ArjunaJTS/trailmap/pom.xml:
```diff
@@ -1,122 +1,248 @@
-        </pluginManagement>
-    </build>
-</project>
+  LLC, and others contributors as indicated by the @authors tag. All rights 
```
@marcosgopen there is a lot of code formatting change here, please can you only include the changed parts?

marcosgopen commented:
Thanks @tomjenkinson for your review. I reset, edit and rebase the commits in order to address your suggestions.

tomjenkinson commented:
@marcosgopen the build output is showing up quite a few problems but I think it could be because the Narayana versions is not updated in the recent release. I am going to try to change that.

tomjenkinson commented:
@marcosgopen I hope I dealt with the version thing in https://github.com/jbosstm/quickstart/commit/7316ce71a1534f34b9682c833672c72f282642a3 - please can you rebase on the master branch?

marcosgopen commented:
Thank you @tomjenkinson , I agree it is a matter of master version name. I rebased the commits on the master as you suggested.

tomjenkinson commented on ArjunaJTS/standalone/README.md:
````diff
@@ -19,19 +19,19 @@ mvn clean compile
 ```
 
-Using the run script will run the example twice, once with JacOrb and then again but using JdkOrb.
+Using the run script will run using OpenJdkOrb.
````
"OpenJDK ORB" rather than "OpenJdkOrb"

tomjenkinson commented on ArjunaJTS/standalone/README.md:
```diff
@@ -19,19 +19,19 @@ mvn clean compile
 remote transaction manager (TM) using a CORBA Name Service for looking up the TM.
-The example runs using either JacOrb or JdkOrb:
+The example runs using OpenJdkOrb:
 
```
"OpenJDK ORB" rather than "OpenJdkOrb"

tomjenkinson commented on ArjunaJTS/standalone/README.md:
```diff
@@ -44,54 +44,40 @@ please override the setting in `start.sh` if it is incorrect.
   the script to Windows batch commands.
 
-Running with JacOrb:
+Running with openjdkorb:
```
"OpenJDK ORB" rather than "openjdkorb"

tomjenkinson replied:
I think this one still needs to be changed to "OpenJDK ORB"?

tomjenkinson commented on ArjunaJTS/standalone/README.md:
````diff
@@ -44,54 +44,40 @@ please override the setting in `start.sh` if it is incorrect.
 1. start a CORBA name server
 ```
-./start.sh jacorb NS
+./start.sh openjdkorb NS
````
Do you think this argument should be required still? I can see it is read here: https://github.com/jbosstm/quickstart/blob/caef497c0b02ce985963de3a7ef64622edaf9416/ArjunaJTS/standalone/start.sh but I wonder if we should change that to not be needed?

tomjenkinson commented on ArjunaJTS/standalone/run.bat:
```diff
@@ -18,11 +18,7 @@ rem MA  02110-1301, USA.
-IF %ERRORLEVEL% NEQ 0 exit -1
-
-echo "Running JTS standalone using JdkORB"
+echo "Running JTS standalone using OpenJdkORB"
```
"OpenJDK ORB" rather than "OpenJdkORB"

tomjenkinson commented on ArjunaJTS/standalone/run.sh:
```diff
@@ -19,26 +19,16 @@
-
-echo "Running jts standalone quickstart using JdkOrb"
-mvn -e exec:java  -Dexec.cleanupDaemonThreads=false -Dexec.mainClass=org.jboss.narayana.jta.quickstarts.TransactionExample $JDKORBPROPS $NARAYANA_VERSION_PARAM
+echo "Running jts standalone quickstart using OpenJdkOrb"
```
"OpenJDK ORB" rather than "OpenJdkOrb" and same a couple of lines down

tomjenkinson commented:
@marcosgopen I made some suggestions on case/spacing name for "OpenJDK ORB" - are these fine for you? I think it could be approved then

marcosgopen commented:
Yes @tomjenkinson thanks! I agree with your suggestions and I rebase the PR. I also added some additional changes related to your suggestion.

tomjenkinson commented on ArjunaJTS/standalone/start.sh:
```diff
@@ -74,17 +58,15 @@ if test "x$HOST_ADDRESS" = "x"; then
 fi
 if [ $# -lt 2 ]; then
-  echo "syntax {jdkorb|jacorb} {NS | TM | RM | CL}"
+  echo "syntax {NS | TM | RM | CL}"
```
I think this is good but I think the test `if [ $# -lt 2 ]; then` should be changed to `if [ $# -lt 1 ]; then` and I think lower down `ORBTYPE=$1` should be removed (and I think `#ORBTYPE=openjdkorb` should be deleted) and I think lower done `case $2 in` should be `case $1 in`

tomjenkinson commented on jta-and-hibernate-standalone/src/test/java/org/jboss/narayana/quickstart/jta/TestCase.java:
```diff
@@ -74,6 +77,8 @@ public class TestCase {
      */
     private Weld weld;
 
+    private static Logger log = LogManager.getLogger(TestCase.class);
```
I think this and the imports can be removed?

tomjenkinson reviewed (approved):
Thank you!
